Mathlib/Data/Nat/Choose/Lucas.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2023 Gareth Ma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gareth Ma
+-/
+import Mathlib.Data.ZMod.Basic
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Lucas's theorem
+
+This file contains a proof of [Lucas's theorem](https://www.wikiwand.com/en/Lucas%27s_theorem) about
+binomial coefficients, which says that for primes `p`, `n` choose `k` is congruent to product of
+`n_i` choose `k_i` modulo `p`, where `n_i` and `k_i` are the base-`p` digits of `n` and `k`,
+respectively.
+
+## Main statements
+
+* `choose_modEq_prod_range_choose`: the binomial coefficient `n choose k` is congruent to the
+product of `n_i choose k_i` modulo `p`, where `n_i` and `k_i` are the base-`p`
+digits of `n` and `k`, respectively.
+-/
+
+open Finset hiding choose
+
+open Nat BigOperators Polynomial
+
+namespace Choose
+
+variable {n k p : ℕ} [Fact p.Prime]
+
+/-- **Lucas's Theorem**: For primes `p`, `choose n k` is congruent to `choose (n % p) (k % p) *
+choose (n / p) (k / p)` modulo `p`. -/
+theorem choose_modEq_choose_mod_mul_choose_div :
```

Comment:
I have a hard time thinking this unambiguously refers to Lucas theorem

Reply:
It's (somewhat trivially) equivalent to the product version, and potentially useful at times, but I can remove the theorem label for it.

Reply:
Oh I meant the lemma name, not the statement